controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,11 +84,439 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+
+	if err != nil {
+		return fmt.Errorf("Error fetching CephFilesystemList. %+v", err)
+	}
+
+	var cephFileSystemName string
+	if len(cephFilesystemList.Items) == 1 {
+		cephFileSystemName = cephFilesystemList.Items[0].Name
+	}
+
+	desired := &rookCephv1.CephFilesystemSubVolumeGroup{
+		Spec: rookCephv1.CephFilesystemSubVolumeGroupSpec{
+			FilesystemName: cephFileSystemName,
+		},
+	}
+
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
+		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
+			return err
+		}
+
+		r.cephFilesystemSubVolumeGroup.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephFilesystemSubVolumeGroup.", "CephFilesystemSubVolumeGroup", klog.KRef(r.cephFilesystemSubVolumeGroup.Namespace, r.cephFilesystemSubVolumeGroup.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.SubVolumeGroupName = r.cephFilesystemSubVolumeGroup.Name
+	return nil
+
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientRBDProvisioner() error {
+
+	// TODO: Update caps relavent to rbd provisioner
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientRBDProvisioner, func() error {
+		if err := r.own(r.cephClientRBDProvisioner); err != nil {
+			return err
+		}
+
+		r.cephClientRBDProvisioner.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBDProvisioner.Namespace, r.cephClientRBDProvisioner.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.CephRBDProvisionerUser = r.cephClientRBDProvisioner.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientRBDNode() error {
+
+	// TODO: Update caps relavent to rbd node
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientRBDNode, func() error {
+		if err := r.own(r.cephClientRBDNode); err != nil {
+			return err
+		}
+
+		r.cephClientRBDNode.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBDNode.Namespace, r.cephClientRBDNode.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.CephRBDNodeUser = r.cephClientRBDNode.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientCephFSProvisioner() error {
+
+	// TODO: Update caps relavent to cephfs provisioner
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
```

Comment:
```suggestion
				"mon": "allow r",
```

Reply:
Noted, this will be handled in a new PR containing cephfs changes